src/main/java/net/sf/jabref/logic/importer/fetcher/MedlineFetcher.java

```diff
@@ -116,45 +118,61 @@ public HelpFile getHelpPage() {
     }
 
     @Override
-    public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
-        String cleanQuery = identifier.trim().replace(';', ',');
+    public URL getURLForID(String identifier) throws URISyntaxException, MalformedURLException, FetcherException {
+        return createFetchUrl(identifier);
+    }
+
+    @Override
+    public Parser getParser() {
+        Parser medlineParser = parserInputStream -> {
```

Comment:
Could you please move the code to the `MedlineImporter` class i.e. implement the `Parser` interface there and here just return `new MedlineImporter()`